src/workerinfo.js

```diff
@@ -0,0 +1,41 @@
+let taskcluster = require('taskcluster-client');
+let assert      = require('assert');
+let debug       = require('debug')('workerinfo');
+
+class WorkerInfo {
+  constructor(options) {
+    assert(options);
+    assert(options.Provisioner);
+    this.Provisioner = options.Provisioner;
+  }
+
+  async provisionerSeen(provisionerId) {
+    let expires = taskcluster.fromNow('5 days');  // temporary hard coded expiration
+
+    let provisioner = await this.Provisioner.load({provisionerId}, true);
```

Comment:
please store the provisionerId as seen in memory for say 12 hours... that way it won't cost much...
Right now you are adding a request to every single `claimWork` request...

This way the overhead of doing this is `N * |provisionerIds| / 12 hours` where `N` is the number of dynos we have running... But the cost won't grow proportional to the number of requests, which is really important, as we have 20k req/min...